rmb-sdk-go/peer/connection.go

```diff
@@ -146,37 +148,57 @@ func (c *InnerConnection) Start(ctx context.Context) (Reader, Writer) {
 
 // listenAndServe creates the websocket connection, and if successful, listens for and serves incoming and outgoing messages.
 func (c *InnerConnection) listenAndServe(ctx context.Context, output chan []byte, input chan []byte) error {
-	con, err := c.connect()
+	connections, err := c.connect()
 	if err != nil {
 		return errors.Wrap(err, "failed to reconnect")
 	}
 
-	return c.loop(ctx, con, output, input)
+	var m sync.Mutex
+	var wg sync.WaitGroup
+
+	for _, con := range connections {
+		wg.Add(1)
+		go func(con *websocket.Conn) {
+			defer wg.Done()
+			m.Lock()
```

Comment:
this will lead to the first connection acquiring the lock and since `c.loop` is a blocking call it won't release the lock and other connections won't be able to start their connection loop. which means it's still a single relay connection used and will switch when an error happen (the lock released).
we can move `c.loop` to be before the lock or imo using channels will coordinate the different goroutines better.